app/Http/Controllers/Api/PublisherController.php

```diff
@@ -31,11 +34,58 @@ public function createPublisher(Request $request)
             }
 
             return response()->json($publisher);
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
             return response()->json(['error' => 'An error occurred while trying to create the publisher'], 500);
         }
     }
 
+    public function viewAllPublishers(Request $request)
+    {
+        try {
+            $perPageOptions = [20, 50, 100];
+
+            $searchValue = $request->input('search', '');
+            $perPage = $request->input('per_page', 20);
+
+            if (!in_array($perPage, $perPageOptions)) {
+                $perPage = 20;
+            }
+
+            $validator = validator()->make($request->all(), [
```

Comment:
@damir-tahirovic Validation should ideally be performed before accessing the request parameters to ensure that the data is valid before you attempt to use it.